### PR TITLE
[YUNIKORN-2599] Certain shim events are not handled by the state machine

### PR DESCRIPTION
**Pull request by pbacsko**
### What is this PR for?
Certain events are not specified in the state machine, but we keep sending them.

### What type of PR is it?
* [x] - Bug Fix
* [ ] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2599

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: pkg/cache/application_state.go [hunk 1]
```diff
@@ -50,13 +50,12 @@ const (
 	KilledApplication
 	ReleaseAppAllocation
 	ReleaseAppAllocationAsk
-	AppStateChange
 	ResumingApplication
 	AppTaskCompleted
 )
 
 func (ae ApplicationEventType) String() string {
-	return [...]string{"SubmitApplication", "RecoverApplication", "AcceptApplication", "TryReserve", "UpdateReservation", "RunApplication", "RejectApplication", "CompleteApplication", "FailApplication", "KillApplication", "KilledApplication", "ReleaseAppAllocation", "ReleaseAppAllocationAsk", "AppStateChange", "ResumingApplication", "AppTaskCompleted"}[ae]
```
Comment (pbacsko) [inline pkg/cache/application_state.go:59]: Note: "RecoverApplication" had to be removed, the string output was actually incorrect.

Comment (chia7712) [reply pkg/cache/application_state.go:59]: Could you please add unit test to keep consistency between enum and string?

**Review by chia7712 — APPROVED** [review]
LGTM

File: pkg/cache/application_state.go (hunk 1 above)
Comment (pbacsko) [reply pkg/cache/application_state.go:59]: Yes, I'll update the PR.

**Pull request closed without merging**